packages/ui-avatar/src/Avatar/index.tsx

```diff
@@ -132,14 +132,16 @@ class Avatar extends Component<Props, State> {
   }
 
   componentDidUpdate(prevProps: Props, prevState: State) {
-    let updateConfig: MakeStyleConfig | undefined
+    let makeStyleConfig: MakeStyleConfig | undefined
     if (prevState.loaded == this.state.loaded) {
-      updateConfig = {
-        prevProps: prevProps,
-        styledProps: ['size', 'shape', 'src']
+      makeStyleConfig = {
+        updateConfig: {
+          prevProps: prevProps,
+          styledProps: ['size', 'shape', 'src']
+        }
       }
     }
-    this.props.makeStyles({ ...updateConfig, ...this.state })
+    this.props.makeStyles({ ...makeStyleConfig, ...this.state })
```

Comment:
Just a note:
You don't have to spread the makeStyleConfig (or even give typing to it) if you would rename it back to `updateConfig` and  then simply do:

```js
let updateConfig
if (prevState.loaded == this.state.loaded) {
    updateConfig = { 
      prevProps,
      styledProps: ['size', 'shape', 'src']
    }
  
}
makeStyles({ updateConfig, ...this.state})
```

This works because TS has "shape" typings, so as long the shape of the object you pass to the function matches the shape of the defined type it will compile correctly. What that means is that you don't have to have a proper `MakeStyleConfig` type variable in order to be able to call `makeStyles`, only an object which has the same shape as of `MakeStyleConfig`.

Reply:
OK, good idea